File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/GetPropertyElementListResponse.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.opcfoundation.webserver.types.digitaltwin.PropertyElementDescriptor;

import java.util.ArrayList;
import java.util.List;

public class GetPropertyElementListResponse {
    private final List<PropertyElementDescriptor> elements;
    private boolean containsMoreData;

    public GetPropertyElementListResponse()
    {
        this.elements = new ArrayList<>();
        this.containsMoreData = false;
    }

    public void add(PropertyElementDescriptor descriptor)
    {
        elements.add(descriptor);
    }

    public List<PropertyElementDescriptor> getElements() {
        return elements;
    }

    public boolean containsMoreData()
    {
        return containsMoreData;
    }

    public void setContainsMoreData(boolean containsMoreData) {
        this.containsMoreData = containsMoreData;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/GetPropertyElementListResponse.ts
import { PropertyElementDescriptor } from "../../../types/digitaltwin/PropertyElementDescriptor";

export class GetPropertyElementListResponse {
    private readonly _elements: Array<PropertyElementDescriptor>;
    private _containsMoreData: boolean;

    constructor() {
        this._elements = [];
        this._containsMoreData = false;
    }

    add(descriptor: PropertyElementDescriptor): void {
        this._elements.push(descriptor);
    }

    get elements(): Array<PropertyElementDescriptor> {
        return this._elements;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }

    set containsMoreData(containsMoreData: boolean) {
        this._containsMoreData = containsMoreData;
    }
}